connman/vpn/plugins/openvpn.c

```diff
@@ -1113,6 +1154,26 @@ static gboolean ov_management_handle_input(GIOChannel *source,
 			 */
 			vpn_provider_add_error(data->provider,
 					VPN_PROVIDER_ERROR_AUTH_FAILED);
+
+			/*
+			 * If the auth error does not terminate the process or
+			 * request anything new, such as plain auth error does,
+			 * it is required to restart the process in order to
+			 * make the connection error detection work and stop
+			 * connecting process when limit is reached. Calling
+			 * ov_connect_done() with EACCES adds auth error, so
+			 * since we need to abort this connection call with
+			 * ECONNABORTED as these errors need user interaction.
+			 * ECONNABORTED also triggers disabling of autoconnect
+			 * in connmand.
+			 */
+			if (error_limit_reached(data->provider) ||
+						is_auth_error_terminal(str)) {
+				ov_connect_done(data, ECONNABORTED);
```

Comment:
Just wondering this that should the terminal errors be regarded as connection errors. In such case there should be either a generic and VPN specific limit, like there is one with authentication errors. It would mean a bit complex bit here, however, these errors in OpenVPN are prefixed with `AUTH`...

Reply:
It might actually make more sense. To check both auth and connection errors. However, some servers seem to report plain 
> openvpn request >PASSWORD:Verification Failed: 'Auth'
when the ciphers are wrong. So this would be auth error and the 
> openvpn request >PASSWORD:Verification Failed: 'Auth' ['Data channel cipher negotiation failed (no shared cipher)']
follows at some point.

Reply:
I think it is enough for now at least. New error counter might be needed, since these are more of configuration errors reported as authentication errors. But it works like this. Need to file a continuation task later on to add a new error type counting.